File: app/src/main/java/org/michaelbel/moviemade/ui/modules/similar/SimilarMoviesActivity.java
package org.michaelbel.moviemade.ui.modules.similar;

import android.os.Bundle;

import org.michaelbel.moviemade.R;
import org.michaelbel.moviemade.data.entity.Movie;
import org.michaelbel.moviemade.ui.base.BaseActivity;
import org.michaelbel.moviemade.utils.IntentsKt;

import androidx.appcompat.widget.AppCompatTextView;
import androidx.appcompat.widget.Toolbar;
import butterknife.BindView;

public class SimilarMoviesActivity extends BaseActivity {

    // TODO make private.
    // TODO make add getter
    public Movie movie;
    private SimilarMoviesFragment fragment;

    @BindView(R.id.toolbar) Toolbar toolbar;
    @BindView(R.id.toolbar_subtitle) AppCompatTextView toolbarSubtitle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_similar);

        movie = (Movie) getIntent().getSerializableExtra(IntentsKt.MOVIE);

        setSupportActionBar(toolbar);
        toolbar.setNavigationOnClickListener(v -> finish());
        toolbar.setOnClickListener(v -> fragment.recyclerView.smoothScrollToPosition(0));
        toolbarSubtitle.setText(movie.getTitle());

        fragment = (SimilarMoviesFragment) getSupportFragmentManager().findFragmentById(R.id.fragment);
        if (fragment != null) {
            fragment.presenter.getSimilarMovies(movie.getId());
        }
    }
}

File: app/src/main/java/org/michaelbel/moviemade/ui/modules/similar/SimilarMoviesActivity.kt
package org.michaelbel.moviemade.ui.modules.similar

import android.os.Bundle
import androidx.appcompat.widget.Toolbar
import kotlinx.android.synthetic.main.activity_subtitle.*
import org.michaelbel.moviemade.R
import org.michaelbel.moviemade.data.entity.Movie
import org.michaelbel.moviemade.ui.base.BaseActivity
import org.michaelbel.moviemade.utils.MOVIE

class SimilarMoviesActivity : BaseActivity() {

    fun getToolbar() : Toolbar {
        return toolbar
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_subtitle)

        val movie = intent.getSerializableExtra(MOVIE) as Movie

        setSupportActionBar(toolbar)
        toolbar!!.setNavigationOnClickListener { finish() }
        toolbar_title.setText(R.string.similar_movies)
        toolbar_subtitle.text = movie.title

        if (savedInstanceState == null) {
            startFragment(SimilarMoviesFragment.newInstance(movie.id), R.id.fragment_view)
        }
    }
}